src/analysis/ODE/picard_lindelof.lean

```diff
@@ -328,3 +328,161 @@ begin
+  Picard-Lindelöf theorem. -/
+lemma time_indep_cont_diff_is_picard_lindelof
+  [proper_space E] (v : E → E) (hv : cont_diff ℝ 1 v) (t₀ : ℝ) (x₀ : E) :
+  ∃ (ε : ℝ) (hε : 0 < ε), is_picard_lindelof (λ t, v) (t₀ - ε) t₀ (t₀ + ε) x₀ :=
+begin
+  have : cont_diff_at ℝ 1 v x₀ := cont_diff.cont_diff_at hv,
+  have hlip := cont_diff_at.exists_lipschitz_on_with this,
+  rcases hlip with ⟨L, s, hs, hlip⟩,
+  rw metric.mem_nhds_iff at hs,
+  rcases hs with ⟨r, hr, hball⟩,
+  have hclosed : metric.closed_ball x₀ (r / 2) ⊆ metric.ball x₀ r,
+  { apply metric.closed_ball_subset_ball,
+    apply half_lt_self,
+    exact hr },
+  have hlip' := lipschitz_on_with.mono hlip (subset_trans hclosed hball),
+  have hbbd : bdd_above ((λ x, ∥v x∥)'' (metric.closed_ball x₀ (r / 2))),
+  { apply is_compact.bdd_above_image,
+    { apply is_compact_closed_ball }, -- uses proper_space E
+    apply continuous_on.norm,
+    apply cont_diff_on.continuous_on (cont_diff.cont_diff_on hv) },
+  rw bdd_above_def at hbbd,
+  cases hbbd with C hC,
+  set ε := if C = 0 then 1 else (r / 2 / C) with hε,
+  use ε,
+  have hε' : 0 < ε,
+  { rw hε,
+    split_ifs,
+    { exact zero_lt_one },
+    rw lt_div_iff,
+    { rw [lt_div_iff zero_lt_two, zero_mul, zero_mul],
+      exact hr,
+      apply_instance },
+    { have : ∥v x₀∥ ≤ C,
+      { apply hC,
+        rw set.mem_image,
+        use x₀,
+        rw [metric.mem_closed_ball, dist_self],
+        refine ⟨_, rfl⟩,
+        rw [le_div_iff zero_lt_two, zero_mul],
+        exact le_of_lt hr,
+        apply_instance },
+      apply lt_of_le_of_ne _ (ne.symm h),
+      exact le_trans (norm_nonneg (v x₀)) this } },
+  use hε',
+  have ht₀ : t₀ ∈ set.Icc (t₀ - ε) (t₀ + ε),
+  { rw ←real.closed_ball_eq_Icc,
+    apply metric.mem_closed_ball_self,
+    apply le_of_lt,
+    exact hε' },
+  -- show is_picard_lindelof
+  refine ⟨L, r / 2, C, _, _, _, _, _⟩,
+  { apply le_of_lt,
+    exact half_pos hr },
+  { intros t ht,
+    exact hlip' },
+  { intros x hx,
+    exact continuous_on_const },
+  { intros t ht x hx,
+    apply hC,
+    rw set.mem_image,
+    use x,
+    rw metric.mem_closed_ball,
+    refine ⟨_, rfl⟩,
+    rw ←metric.mem_closed_ball,
+    exact hx },
+  { rw [add_tsub_cancel_left, sub_sub_cancel, max_self, hε],
+    split_ifs,
+    { rw [h, zero_mul, le_div_iff zero_lt_two, zero_mul],
+      exact le_of_lt hr,
+      apply_instance },
+    rw mul_div_cancel' _ h }
+end
+
+/-- A time-independent, continuously differentiable ODE admits a solution in some open interval. -/
+theorem ODE_solution_exists.at_ball_of_cont_diff
+  [proper_space E] (v : E → E) (hv : cont_diff ℝ 1 v) (t₀ : ℝ) (x₀ : E) :
+  ∃ (ε : ℝ) (hε : 0 < ε) (f : ℝ → E), f t₀ = x₀ ∧
+    ∀ t ∈ metric.ball t₀ ε, has_deriv_at f (v (f t)) t :=
+begin
+  have h := time_indep_cont_diff_is_picard_lindelof v hv t₀ x₀,
+  rcases h with ⟨ε, hε, hpl⟩,
+  use ε,
+  use hε,
+  apply ODE_solution_exists.at_ball (λ t, v) t₀ ε hε x₀ hpl
```

Comment:
Again here you can use `obtain` and then `  exact ⟨ε, hε, ODE_solution_exists.at_ball (λ t, v) t₀ ε hε x₀ hpl⟩`.